cores/esp8266/HardwareSerial.cpp

```diff
@@ -94,6 +97,9 @@ void HardwareSerial::setDebugOutput(bool en)
     if(en) {
         if(uart_tx_enabled(_uart)) {
             uart_set_debug(_uart_nr);
+#if !defined(DEBUG_ESP_PORT) && !defined(NDEBUG)
+            os_printf("%s\r\n", ESP.getFullVersion().c_str());
+#endif

```

Comment:
I'm fine with printing the version in Serial::begin if debug output is enabled... however if i'm calling setDebugOutput i probably know what i'm doing and can handle calling ESP.getFullVersion if needed.

Reply:
Removing this one will also allow the linker to get rid of the code if sources are moved to a separate file.